### PR TITLE
Install elasticsearch earlier in travis and retry connecting up to 3 times

### LINKED ISSUE #14828: Upgrade travis to use Ubuntu 18.04 LTS
Travis currently uses 16.04 LTS.
Let's update to 18.04 which is the latest LTS supported by travis, because why not. All tests pass.

### LINKED ISSUE #15077: port 9200: Connection refused intermittent
I've seen this a few times on Travis. [This time](https://travis-ci.org/github/mozilla/addons-server/jobs/712253479) was in the ES job, but I'm not sure if that's always been the case.

Console messages:
```
$ if [ $TOXENV == "es" ]; then
      sleep 10;
      curl -v http://localhost:9200/;
  fi
  
*   Trying 127.0.0.1...
* connect to 127.0.0.1 port 9200 failed: Connection refused
* Failed to connect to localhost port 9200: Connection refused
* Closing connection 0
curl: (7) Failed to connect to localhost port 9200: Connection refused
The command "if [ $TOXENV == "es" ]; then
      sleep 10;
      curl -v http://localhost:9200/;
  fi
  " failed and exited with 7 during .

Your build has been stopped.
```

### PR DESCRIPTION
**Pull request by diox**
Might help to give a tiny bit more time for ES to start.

Fixes #15077
Fixes #14828 (needed for more recent version of `curl`)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -82,8 +82,7 @@ before_script:
   - java -version
   - |
     if [ $TOXENV == "es" ]; then
-        sleep 10;
-        curl -v http://localhost:9200/;
+        curl --retry 3 --retry-delay 10 --retry-connrefused http://localhost:9200/;
```
Comment (diox) [inline .travis.yml:85]: `curl: option --retry-connrefused: is unknown` :disappointed: 

It does exist in our image, but travis does not quite match that debian version...

Comment (diox) [reply .travis.yml:85]: https://github.com/mozilla/addons-server/pull/15115/commits/c4fc80877af18a57c1236f05ce21e7ed2094f43d should fix that

File: .travis.yml [hunk 2]
```diff
@@ -68,6 +58,16 @@ before_install:
...
         sudo systemctl start elasticsearch
     fi
+  - mysql -e 'create database olympia;'
+  - export GOPATH=$HOME/go
+  - export PATH=$HOME/usr/local/go/bin:$GOPATH/bin:$PATH
+  # Unfortunately we require the most recent libmagic-dev version to make use of more recent
+  # JSON detection features which we require for the code-manager related APIs
+  - echo "deb http://archive.ubuntu.com/ubuntu eoan main" | sudo tee -a /etc/apt/sources.list
+  - sudo apt-get update -qq
+  - sudo apt-get install -t eoan libmagic-dev libmagic1 libmagic-mgc
```
Comment (willdurand) [inline .travis.yml:68]: I am wondering if we still need all of that 🤔

Comment (diox) [reply .travis.yml:68]: My bad, you removed it in 09bbe0a61661dfb53927c3ab42af0fddc38614d8 but I accidentally added it back when merging ES 6 support in ec0d8f9293d8600afffcf9136a2c0875197846cd - so when I did this PR I moved it without realizing it should have been removed already...

Comment (diox) [reply .travis.yml:68]: https://github.com/mozilla/addons-server/pull/15157

**Pull request merged**